ads/google/a4a/utils.js

```diff
@@ -441,6 +440,25 @@ export function getCorrelator(win, nodeOrDoc, opt_cid) {
   return win.ampAdPageCorrelator;
 }
 
+/**
+ * @param {(string|undefined)} clientId
+ * @param {string} pageViewId
+ * @return {number}
+ */
+function makeCorrelator(clientId, pageViewId) {
+  const pageViewIdNumeric = Number(pageViewId || 0);
+  if (clientId) {
+    return pageViewIdNumeric + ((clientId.replace(/\D/g, '') % 1e6) * 1e6);
```

Comment:
I realize you're only moving code but a couple of things:

- You don't need pageViewIdNumeric if clientId is falsey
- You don't need an else clause
- What happens if pageViewId is NaN?  Same for clientId?

I would change this to (assumes clientId and pageViewId are integers, if not can change to parseFloat):

if (!isNaN(parseInt(clientId, 10))) {
   return (parseInt(pageViewId, 10) || 0) + ((clientId.replace(/\D/g, '') % 1e6) * 1e6);
}
return Math.floor...

Would be ideal to wrap in an experiment to see if this makes a change.  Also there is no test coverage for this code... please add some.